systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/KafkaClientProperties.java

```diff
@@ -122,6 +373,60 @@ static Properties createConsumerProperties(String namespace, String clusterName,
+                                "oauth.client.secret=\"" + clientSecretName + "\" " +
+                                "oauth.token.endpoint.uri=\"" + oauthTokenEndpointUri + "\";");
+            } else {
+                // tls
+                String responseKeycloak = Exec.exec("openssl", "s_client", "-showcerts", "-connect", kubeClient().getNodeAddress() + ":" + Constants.HTTPS_KEYCLOAK_DEFAULT_NODE_PORT).out();
+                Matcher matcher = Pattern.compile("-----(?s)(.*)-----").matcher(responseKeycloak);
+
+                if (matcher.find()) {
+                    String keycloakCertificateData = matcher.group(0);
+                    LOGGER.info("Keycloak cert is:{}\n", keycloakCertificateData);
+
+                    LOGGER.info("Creating keycloak.crt file");
+                    File keycloakCertFile = File.createTempFile("keycloak", ".crt");
+
+                    Files.write(keycloakCertFile.toPath(), keycloakCertificateData.getBytes(StandardCharsets.UTF_8));
+
+                    LOGGER.info("Importing keycloak certificate {} to truststore", keycloakCertFile.getAbsolutePath());
+                    Exec.exec("keytool", "-v", "-import", "-trustcacerts", "-file", keycloakCertFile.getAbsolutePath(),
+                            "-alias", "keycloakCrt1", "-keystore", consumerProperties.get("ssl.truststore.location").toString(),
+                            "-noprompt", "-storepass", consumerProperties.get("ssl.truststore.password").toString());
+                }
+
+
+                LOGGER.info("Setting oauth properties");
+
+                LOGGER.info("Enabling SSL with setting {}={}", CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SASL_SSL");
+                consumerProperties.setProperty(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SASL_SSL");
+                consumerProperties.setProperty("sasl.mechanism", "OAUTHBEARER");
+                consumerProperties.setProperty("sasl.login.callback.handler.class", "io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler");
+                consumerProperties.setProperty("sasl.jaas.config",
+                        "org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule " +
+                                "required " +
+                                "oauth.client.id=\"" + clientId + "\" " +
+                                "oauth.client.secret=\"" + clientSecretName + "\" " +
+                                "oauth.token.endpoint.uri=\"" + oauthTokenEndpointUri + "\" " +
+                                "oauth.ssl.endpoint.identification.algorithm=\"\"" +
+                                "oauth.ssl.truststore.location=\"" + consumerProperties.get("ssl.truststore.location") + "\" " +
+                                "oauth.ssl.truststore.password=\"" + consumerProperties.get("ssl.truststore.password") + "\" " +
+                                "oauth.ssl.truststore.type=\"" + consumerProperties.get("ssl.truststore.type") + "\" ;");
+            }
```

Comment:
it seems to be the same code as the producer, I would suggest to factor out a common method for it.